corehq/messaging/scheduling/scheduling_partitioned/models.py

```diff
@@ -679,3 +679,13 @@ class Meta(AbstractTimedScheduleInstance.Meta):
         unique_together = (
             ('case_id', 'timed_schedule_id', 'recipient_type', 'recipient_id'),
         )
+
+    def additional_deactivation_condition_reached(self):
+        return_condition = super().additional_deactivation_condition_reached()
+
+        from corehq.apps.data_interfaces.models import AutomaticUpdateRule
```

Comment:
the import should be at the top of the scope, here the function.

Reply:
fixed. [b5564a4](https://github.com/dimagi/commcare-hq/pull/32671/commits/b5564a469edfb0237e348d306ce1e44e5a7450c7)